packages/cozy-intent/src/api/services/NativeService.ts

```diff
@@ -21,14 +21,30 @@ export class NativeService {
   public registerWebview = (webviewRef: WebviewRef): void => {
     const uri = StaticService.getUri(webviewRef)
 
-    if (this.messengerRegister[uri]) return
+    if (this.messengerRegister[uri]) {
+      throw new Error(
+        `Cannot register webview. A webview is already registered into cozy-intent with the uri: ${uri}`
+      )
+    }
 
     this.messengerRegister = {
       ...this.messengerRegister,
       [uri]: { messenger: new this.messengerService(webviewRef) }
     }
   }
 
+  public unregisterWebview = (webviewRef: WebviewRef): void => {
+    const uri = StaticService.getUri(webviewRef)
+
+    if (!this.messengerRegister[uri]) {
+      throw new Error(
+        `Cannot unregister webview. No webview is registered into cozy-intent with the uri: ${uri}`
+      )
+    }
+
+    delete this.messengerRegister[uri]
```

Comment:
could we add the simple tests to document the feature and prevent regressions?

Reply:
I think this can be done, as long as we don't write full integration test simulating mobile and dom env (which would be doable but difficult, and would mostly test post-me and not our code)

Reply:
I meant Unit Testing:

stubbing StaticService.getUri to return one uri
stubbing messengerRegister to return a boolean, and expect either throw or delete 
and calling unregisterWebview

Reply:
Yeah sure, I'm all for unit testing the public API of this lib, as long as we don't test `post-me` itself. Though these tests should have already been written by me. @Ldoppea do you have time to do it for this method? If not, I will do it myself on Friday afternoon

Reply:
@acezard I can do it, but there is no test script configured in the package.json. Did you plan to use a specific framework?

Reply:
Not really just what cozy uses. Though I would have liked to write the test in typescript files. spec.ts

Reply:
I added tests for `registerWebview` and `unregisterWebview`.

`unregisterWebview` was pretty tricky as we cannot test the private `messengerRegister` so we had to test the `tryEmit` to verify that it fails. Not the most direct test but at least we have something.

Thanks @acezard for the help.